From 3321ed0ff53e2b9f2a2c63f16696bb784ac27f14 Mon Sep 17 00:00:00 2001
From: bxu10x <bingx.xu@intel.com>
Date: Wed, 5 Dec 2018 09:14:36 +0800
Subject: [PATCH] DO NOT MERGE Expose the viewHolder.toggle of
 IconToggleLineItem
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Expose the viewHolder.toggle of IconToggleLineItem if it is needed to be
changed. The toggle icon can be enable/disable by others.

Bug: 114013494
Test: 1) Launch “com.android.cts.verifier” apk
      2) Start test: BYOD Managed Provisioning -> Permissions lockdown ->
         Choose “Grant” or “Deny”

Change-Id: Iad0d49aff6a28bf7ad24842924813ad757184314
---
 car-list/src/com/android/car/list/IconToggleLineItem.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/car-list/src/com/android/car/list/IconToggleLineItem.java b/car-list/src/com/android/car/list/IconToggleLineItem.java
index 75100be..6b11982 100644
--- a/car-list/src/com/android/car/list/IconToggleLineItem.java
+++ b/car-list/src/com/android/car/list/IconToggleLineItem.java
@@ -39,6 +39,7 @@ public abstract class IconToggleLineItem
     private final CharSequence mTitle;
     protected IconUpdateListener mIconUpdateListener;
     protected SwitchStateUpdateListener mSwitchStateUpdateListener;
+    private Switch mSwitchToggle;
 
     /**
      * Interface for enabling icon to be changed by the LineItem.
@@ -95,6 +96,7 @@ public abstract class IconToggleLineItem
         viewHolder.toggle.setChecked(isChecked());
         viewHolder.onUpdateIcon(getIcon());
         viewHolder.toggle.setOnTouchListener(mOnTouchListener);
+        mSwitchToggle = viewHolder.toggle;
         mIconUpdateListener = viewHolder;
         mSwitchStateUpdateListener = viewHolder;
     }
@@ -156,4 +158,8 @@ public abstract class IconToggleLineItem
     public boolean isClickable() {
         return true;
     }
+
+    public Switch getSwitch() {
+        return mSwitchToggle;
+    }
 }
-- 
1.9.1

